### PR TITLE
HBASE-23024 : Replace initcause with Constructor arg

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   2m 58s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 33s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 48s |  master passed  |
| :green_heart: |  compile  |   1m 43s |  master passed  |
| :green_heart: |  checkstyle  |   2m 31s |  master passed  |
| :green_heart: |  shadedjars  |   4m 56s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 22s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 22s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   6m 16s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m 29s |  the patch passed  |
| :green_heart: |  compile  |   1m 45s |  the patch passed  |
| :green_heart: |  javac  |   1m 45s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 31s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  17m 17s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 17s |  the patch passed  |
| :green_heart: |  findbugs  |   6m 46s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   2m 55s |  hbase-common in the patch passed.  |
| :green_heart: |  unit  |   1m 49s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 233m  5s |  hbase-server in the patch failed.  |
| :green_heart: |  asflicense  |   1m 14s |  The patch does not generate ASF License warnings.  |
|  |   | 312m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/627 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 521d67797323 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-627/out/precommit/personality/provided.sh |
| git revision | master / 1cb4f687c1 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/1/testReport/ |
| Max. process+thread count | 4748 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
Please review @petersomogyi @Reidddddd

**Comment by virajjasani** [comment]
Thank you @petersomogyi 
I have tried to include all possible Exceptions. According to me, mostly all are covered. Thanks for the review.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/DroppedSnapshotException.java [hunk 1]
```diff
@@ -40,4 +40,15 @@ public DroppedSnapshotException() {
...
+  /**
+   * DroppedSnapshotException with cause
+   *
+   * @param message the message for this exception
+   * @param cause the cause for this exception
+   */
+  public DroppedSnapshotException(String message, Throwable cause) {
+    super(message, cause);
+  }
+
```
Comment (HorizonNet) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/DroppedSnapshotException.java]: NIT: Could you please remove the empty line (also in the other files).

Comment (virajjasani) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/DroppedSnapshotException.java]: Done

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   2m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 47s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   8m 44s |  master passed  |
| :green_heart: |  compile  |   2m 22s |  master passed  |
| :green_heart: |  checkstyle  |   2m 52s |  master passed  |
| :green_heart: |  shadedjars  |   6m  9s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 50s |  master passed  |
| :blue_heart: |  spotbugs  |   5m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   7m 28s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   7m 11s |  the patch passed  |
| :green_heart: |  compile  |   2m 16s |  the patch passed  |
| :green_heart: |  javac  |   2m 16s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 44s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m 11s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  17m 27s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  javadoc  |   1m 18s |  the patch passed  |
| :green_heart: |  findbugs  |   6m 41s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   2m 55s |  hbase-common in the patch passed.  |
| :green_heart: |  unit  |   1m 48s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 258m 32s |  hbase-server in the patch failed.  |
| :green_heart: |  asflicense  |   1m 30s |  The patch does not generate ASF License warnings.  |
|  |   | 348m 23s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAsyncNonMetaRegionLocatorConcurrenyLimit |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/627 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux bf400379e2bd 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-627/out/precommit/personality/provided.sh |
| git revision | master / 1cb4f687c1 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/2/testReport/ |
| Max. process+thread count | 4814 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-client hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-627/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**